tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -74,28 +118,85 @@ struct instantiator {
                 // Compare sentinel / sentinel
                 assert(Cit{Sen{input}} == Cit{Sen{input}});
                 assert(Cit{Sen{input}} == Cit{Sen{input + 1}});
+
+                if constexpr (CanDifference<Iter>) {
+                    // Difference iterator / iterator
+                    const same_as<iter_difference_t<Iter>> auto diff_it_it = Cit{Iter{input}} - Cit{Iter{input + 1}};
+                    assert(diff_it_it == -1);
+
+                    // Difference iterator / sentinel
+                    const same_as<iter_difference_t<Iter>> auto diff_it_sen = Cit{Iter{input + 1}} - Cit{Sen{input}};

```

Comment:
Despite that our test types provide stronger guarantees that make this perfectly well-defined, these give me icky feelings since `[Iter{input + 1}, Sen{input})` doesn't denote a range ([[iterator.concept.sizedsentinel]/2](http://eel.is/c++draft/iterator.concept.sizedsentinel#2)). Could we move the `+ 1` from the iterator to the sentinel for the cases that have both? (And switch the sign of the results, of course.)